pyspades/player.py

```diff
@@ -336,6 +338,17 @@ def on_input_data_recieved(self, contained: loaders.InputData) -> None:
         z_vel = world_object.velocity.z
         if contained.jump and not (z_vel >= 0 and z_vel < 0.017):
             contained.jump = False
+        # Do not let jump if the player is in the air
+        check = self.protocol.map.get_solid
+        pos = world_object.position
+        oz = 2 + int(not contained.crouch) # TODO CHECCK THIS        
+        if contained.jump and not (
+        check(pos.x+0.5, pos.y+0.5, pos.z + oz) or
+        check(pos.x+0.5, pos.y-0.5, pos.z + oz) or 
+        check(pos.x-0.5, pos.y+0.5, pos.z + oz) or 
+        check(pos.x-0.5, pos.y-0.5, pos.z + oz)):
+            print("WARNING: Attempt to take off. " + str(self.name))
```

Comment:
Don't use print, do use the built in logging. Also, use string formatting instead of +